salt/_pillar/metalk8s_nodes.py

```diff
@@ -92,9 +96,40 @@ def ext_pillar(minion_id, pillar, kubeconfig):
             for node in node_list.items
         )
 
+        custom_object = []
+
+        try:
+            object_response = customObjectsApi.list_cluster_custom_object(
+                group="storage.metalk8s.scality.com", version="v1alpha1",
+                plural="volumes"
+            )
+            if not object_response:
+                custom_object = __utils__['pillar_utils.errors_to_dict']([
+                    'Unable to retrieve node volume list'
+                ])
+            for node in node_list.items:
```

Comment:
Maybe I'm missing something, but how is the volume list dispatched between nodes?

Normally each node should only have its volumes in its pillar, is it done here?

Reply:
Indeed no it's not done. You should not do a `for` loop and use the `minion_id` to only retrieve the node you need

Reply:
Hmmm, I see your point. I can add something like

`if items['spec']['nodeName'] == node:`

We previously had this check.

Reply:
@Ebaneck actually I think you don't need to filter the list yourself, I think the `list_cluster_custom_object` allows you to pass a `label_selector` which you can use to filter the returned result (filter on `spec.nodeName=<minion_id>`)

Reply:
Looks like the current kubernetes python client does not support filters based on `field_selector` for the custom object type. so we are marking this comment as resolved.

Resolution is to filter based on `items['spec']['nodeName'] == <minion_id>:`